lineapy/api/api.py

```diff
@@ -140,43 +141,71 @@ def save(reference: object, name: str) -> LineaArtifact:
     return linea_artifact
 
 
-def delete(
-    artifact_name: str, version: Optional[Union[int, str]] = None
-) -> None:
+def delete(artifact_name: str, version: Union[int, str]) -> None:
     """
     Deletes an artifact from artifact store. If no other artifacts
     refer to the value, the value is also deleted from both the
     value node store and the pickle store.
 
-    If version is not provided, latest version is used.
+    :param artifact_name: Key used to while saving the artifact
+    :param version: version number or 'latest' or 'all'
     """
+    if not is_artifact_version_valid(version):
+        raise ValueError(
+            f"Version {version} is not a valid version.\n"
+            + "Version must be a positive whole number or 'all' or 'latest'."
+        )
+
+    # if version is an integer string, cast to int
+    if version not in ["all", "latest"]:
+        version = int(version)
+
+    # get database instance
     execution_context = get_context()
     executor = execution_context.executor
     db = executor.db
 
-    get_version = None if not isinstance(version, int) else version
-    artifact = db.get_artifact_by_name(artifact_name, version=get_version)
+    # if version is 'all' or 'latest', get_version is None
+    get_version = None
+    if isinstance(version, int):
+        get_version = version
+
+    try:
+        artifact = db.get_artifact_by_name(artifact_name, version=get_version)
```

Comment:
We had determined that delete should not take None due to safety reasons. If the user runs code
from another library that returned None instead of throwing an error, they could unknowingly delete an artifact.

The other functions in both the API and DB accept None, which I personally disagree with. I did not touch those
functions as it is out of scope of this PR.